src/lib.mo

```diff
@@ -139,11 +140,15 @@ module {
     /// ```
     public func addGauge(prefix : Text, buckets : ?[Nat]) : GaugeInterface {
       let id = values.size();
-      let value = GaugeValue(prefix, buckets, now, watermarkResetIntervalSeconds);
+      switch (buckets) {
+        case (?b) ignore Array.foldLeft<Nat, Nat>(b, 0, func(prev, new) = if (prev <= new) { new } else { Debug.trap("Buckets have to be ordered") });
```

Comment:
`prev < new`
"Buckets have to be ordered and non-empty"

Reply:
As discussed in private conversation, the logic remains the same